lupa/_lupa.pyx

```diff
@@ -1207,11 +1261,21 @@ cdef int py_to_lua(LuaRuntime runtime, lua_State *L, object o, bint wrap_none=Fa
         lua.lua_pushnumber(L, <lua.lua_Number>cpython.float.PyFloat_AS_DOUBLE(o))
         pushed_values_count = 1
     elif isinstance(o, long):
-        lua.lua_pushnumber(L, <lua.lua_Number>cpython.long.PyLong_AsDouble(o))
-        pushed_values_count = 1
+        try:
+            lua.lua_pushinteger(L, <lua.lua_Integer>o)
+            pushed_values_count = 1
+        except OverflowError:
+            pushed_values_count = py_to_lua_handle_overflow(runtime, L, o)
+            if pushed_values_count <= 0:
+                raise
     elif IS_PY2 and isinstance(o, int):
-        lua.lua_pushnumber(L, <lua.lua_Number><long>o)
-        pushed_values_count = 1
+        try:
+            lua.lua_pushinteger(L, <lua.lua_Integer>o)
+            pushed_values_count = 1
+        except OverflowError:
+            pushed_values_count = py_to_lua_handle_overflow(runtime, L, o)
+            if pushed_values_count <= 0:
+                raise
```

Comment:
This is entirely the same implementation now, right? Just replace the condition with `elif isinstance(o, (long, int)):`.